### PR TITLE
feat: table viz improvements

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #2531 
Closes: #2482 

### Description:

Screenshot 2022-07-08 at 15 42 33

Screenshot 2022-07-08 at 15 42 54

### CODE REVIEW

File: packages/frontend/src/components/LightdashVisualization/index.tsx [hunk 1]
```diff
@@ -8,19 +8,16 @@ import { useVisualizationContext } from './VisualizationProvider';
...
-                return <SimpleStatistic />;
-            case ChartType.TABLE:
-                return <SimpleTable />;
-            case ChartType.CARTESIAN:
-                return <SimpleChart />;
-            default:
-                return null;
-        }
-    };
-    return <>{renderType()}</>;
```
Comment (rephus) [inline packages/frontend/src/components/LightdashVisualization/index.tsx:23]: just curious, why did you change this ?

File: packages/frontend/src/components/SimpleTable/SimpleTable.styles.tsx [hunk 2]
```diff
@@ -1,25 +1,6 @@
...
-    background: ${Colors.GRAY4};
-`;
-
-export const TableRow = styled.tr<{ i: number }>`
-    background-color: ${({ i }) =>
-        i % 2 ? Colors.LIGHT_GRAY5 : Colors.LIGHT_GRAY4};
-`;
-
-export const TableCell = styled.td<{ isNaN: boolean }>`
-    text-align: ${({ isNaN }) => (isNaN ? 'left' : 'right')} !important;
```
Comment (rephus) [inline packages/frontend/src/components/SimpleTable/SimpleTable.styles.tsx:24]: is this text-align still working  ?

File: packages/frontend/src/components/SimpleTable/index.tsx [hunk 3]
```diff
@@ -1,80 +1,29 @@
-import { HTMLTable, NonIdealState } from '@blueprintjs/core';
-import { getResultValues, isNumericItem } from '@lightdash/common';
+import { getResultValues } from '@lightdash/common';
 import React, { FC } from 'react';
+import Table from '../common/Table';
```
Comment (rephus) [inline packages/frontend/src/components/SimpleTable/index.tsx:3]: nice

**Comment by rephus** [comment]
On dashboards it shoudl always scroll, right ? 
image

**Review by rephus — APPROVED** [review]
Looks good, nice refactoring

**Comment by rephus** [comment]
Mm, something is wrong with the CSV export , not sure if related... 
[lightdash-export-2022-07-11.csv](https://github.com/lightdash/lightdash/files/9081436/lightdash-export-2022-07-11.csv)

image

**Comment by rephus** [comment]
Total number style is different

image

**Comment by rephus** [comment]
By default, the column ordering is not the same 

image

File: packages/frontend/src/components/SimpleTable/SimpleTable.styles.tsx (hunk 2 above)
Comment (ZeRego) [reply packages/frontend/src/components/SimpleTable/SimpleTable.styles.tsx:24]: yes :) handled by the table component based on the column `meta.item`

File: packages/frontend/src/components/LightdashVisualization/index.tsx (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/components/LightdashVisualization/index.tsx:23]: Just removing redundant code and because calling a function inside the jsx is a bad practice.

**Comment by ZeRego** [comment]
@rephus 

> On dashboards it shoudl always scroll, right ?

yes, that was decided after I opened the PR. Pushed a change to have only scroll in the table viz. 

> Mm, something is wrong with the CSV export ...

Wasn't able to replicate the issue. Do you have a recording or steps to reproduce ? 

> by default the column order is not the same as the resultstable

Fixed the issue :) nice catch.

**Comment by rephus** [comment]
> Wasn't able to replicate the issue. Do you have a recording or steps to reproduce ?

Go to saved dashboard, `explore from here` on table chart. Download CSV on table chart (not results, that looks ok) 
Peek 2022-07-11 12-17

**Comment by ZeRego** [comment]
> > Wasn't able to replicate the issue. Do you have a recording or steps to reproduce ?
> 
> Go to saved dashboard, `explore from here` on table chart. Download CSV on table chart (not results, that looks ok) Peek 2022-07-11 12-17 [ ](https://user-images.githubusercontent.com/1983672/178243081-2b707f02-8ad6-4140-a352-407f0c798306.gif) Peek 2022-07-11 12-17 [ ](https://user-images.githubusercontent.com/1983672/178243081-2b707f02-8ad6-4140-a352-407f0c798306.gif)

Opened a separate issue for that #2672

**Review by rephus — APPROVED** [review]
